### ISSUE TITLE
User sign up

### ISSUE THREAD

**Issue by theianjones** [theianjones/remix-discussions#1]
As someone that has just discovered remix-discussions, I would like to sign up for an account.

- create the prisma models
  - viewer
  - session
  - token
- create account creation/log in
  - login page
  - sign up page

**Comment by theianjones** [comment]
@blitzjs has a great template for users, sessions and tokens:

https://github.com/blitz-js/blitz/blob/f2d814813f4860907cb1468ec4a9fa31db34edf9/packages/generator/templates/app/db/schema.prisma#L15-L58
```prisma
model User {
  id             Int      @id @default(autoincrement())
  createdAt      DateTime @default(now())
  updatedAt      DateTime @updatedAt
  name           String?
  email          String   @unique
  hashedPassword String?
  role           String   @default("USER")

  tokens   Token[]
  sessions Session[]
}


model Session {
  id                 Int       @id @default(autoincrement())
  createdAt          DateTime  @default(now())
  updatedAt          DateTime  @updatedAt
  expiresAt          DateTime?
  handle             String    @unique
  hashedSessionToken String?
  antiCSRFToken      String?
  publicData         String?
  privateData        String?

  user   User? @relation(fields: [userId], references: [id])
  userId Int?
}


model Token {
  id          Int      @id @default(autoincrement())
  createdAt   DateTime @default(now())
  updatedAt   DateTime @updatedAt
  hashedToken String
  type        String
  // See note below about TokenType enum
  // type        TokenType
  expiresAt   DateTime
  sentTo      String

  user   User @relation(fields: [userId], references: [id])
  userId Int

  @@unique([hashedToken, type])
}
```

**Comment by theianjones** [comment]
I also found this [remix-auth](https://github.com/sergiodxa/remix-auth) package that looks like its getting some attention. It is alot like `passport.js` but based on web fetch api like remix champions. 

Another option would be to use some oauth service like supabase or Auth0.

**Comment by theianjones** [comment]
[remix cookies](https://remix.run/docs/en/v1/api/remix#cookies) and [remix sessions](https://remix.run/docs/en/v1/api/remix#sessions) are important to consider when building auth.

**Comment by theianjones** [comment]
Going a cookie based sessions strategy seems like the least complicated. This means we dont need a session model in our database.

**Comment by theianjones** [comment]
resolved by #4
